src/components/chat/Message.js

```diff
@@ -7,6 +9,9 @@ class Message extends Component {
     let { message, username } = this.props;
     let fromMe = message.from === username;
     let messageClass = fromMe ? 'chat-outgoing' : 'chat-incoming';
+    
+    let RFCDate = message.key.split('-')[0];
+    let date = formatRelative(Date.parse(RFCDate), new Date());
```

Comment:
@Spetastian This will create a new `Date` object for every message on every render rather than creating a `Date` object once for each message _ever_ -- rather than on every render -- and attaching it to that message.

So we can create a new `Date` upstream and attach it to each message, or...

I'm thinking we just wait till after Nov 4th to integrate this in, since shortly thereafter we'll be getting a timestamp from the server anyway.  But using the UI that @iamrobinpham made should work well :-).

Reply:
@elimisteve Did you meant to ping me here?

Reply:
Yes, since I thought that you'd be the right person to attach the `Date` to each message... but just waiting a bit longer on this is probably better.

Thoughts?

Reply:
Yeah, i would suggest that the date field is created in ChatHandler when parsing the message. So that this component just uses that property as is, no questions asked.

Reply:
@elimisteve Does that make sense?